node/src/accountant/pending_payable_dao.rs

```diff
@@ -25,37 +26,62 @@ pub enum PendingPayableDaoError {
 }
 
 pub trait PendingPayableDao {
-    fn fingerprint_rowid(&self, transaction_hash: H256) -> Option<u64>;
+    fn fingerprints_rowids(&self, hashes: &[H256]) -> Vec<(Option<u64>, H256)>;
     fn return_all_fingerprints(&self) -> Vec<PendingPayableFingerprint>;
-    fn insert_new_fingerprint(
+    fn insert_new_fingerprints(
         &self,
-        transaction_hash: H256,
-        amount: u128,
-        timestamp: SystemTime,
+        hashes_and_amounts: &[(H256, u128)],
+        batch_wide_timestamp: SystemTime,
     ) -> Result<(), PendingPayableDaoError>;
-    fn delete_fingerprint(&self, id: u64) -> Result<(), PendingPayableDaoError>;
-    fn update_fingerprint(&self, id: u64) -> Result<(), PendingPayableDaoError>;
-    fn mark_failure(&self, id: u64) -> Result<(), PendingPayableDaoError>;
+    fn delete_fingerprints(&self, ids: &[u64]) -> Result<(), PendingPayableDaoError>;
+    fn update_fingerprints(&self, ids: &[u64]) -> Result<(), PendingPayableDaoError>;
+    fn mark_failures(&self, ids: &[u64]) -> Result<(), PendingPayableDaoError>;
 }
 
 impl PendingPayableDao for PendingPayableDaoReal<'_> {
-    fn fingerprint_rowid(&self, transaction_hash: H256) -> Option<u64> {
-        let mut stm = self
+    fn fingerprints_rowids(&self, hashes: &[H256]) -> Vec<(Option<u64>, H256)> {
```

Comment:
If rusqlite allows you to use a single question mark in an "in" clause to stand for a whole collection, that would be simpler than using "or" like this. If it doesn't, there's no reason to use "in" here.

However, maybe put a comment in here about how in the future it might be the place to insert some batching logic in case we get a `hashes` collection that's too big to put in a `where` clause with lots of `or`s.